### PR TITLE
Java

### PR DESCRIPTION
**Pull request by markus2330**
## Basics

These points need to be fulfilled for every PR:

- [X] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [X] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [X] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy.
- [X] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -116,9 +116,9 @@ mac_task:
         PLUGINS: ALL;-curlget # The curlget tests fail: https://github.com/ElektraInitiative/libelektra/issues/3382
 
   install_script:
-    - > # Install Homebrew casks
-      brew cask install java oclint
     - | # Install Homebrew formulas
+      brew install openjdk
```
Comment (sanssecours) [inline .cirrus.yml:120]: This will most probably not work. You need to use `brew cask install openjdk` here, and `brew cask install oclint` below.

Comment (markus2330) [reply .cirrus.yml:120]: Yes, it did not work. I thought cask is for GUI? I now tried with a tap.

Comment (sanssecours) [reply .cirrus.yml:120]: >  I thought cask is for GUI?

That is the case a lot of the time. More generally though Homebrew Cask is a place for (non-open) software that already provides  an installer (`.pkg`) or contains a normal (GUI) application (`.app`). Seems like the Homebrew people are actively working on their infrastructure. Now

```sh
brew install openjdk
```

works and

```sh
brew cask install openjdk
```

(which – as far as I remember – worked a few hours ago)  does not any more.

**Comment by markus2330** [comment]
Now there is some other brew error :cry: 

```

brew tap homebrew/services
==> Tapping homebrew/services
Cloning into '/usr/local/Homebrew/Library/Taps/homebrew/homebrew-services'...
Tapped 1 command (66 files, 72.6KB).
brew update # Work around for [Homebrew Services issue 206](https://github.com/Homebrew/homebrew-services/issues/206)
fatal: unable to access 'https://github.com/Homebrew/homebrew-core/': transfer closed with outstanding read data remaining
Error: Fetching /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core failed!
```

**Comment by markus2330** [comment]
Pushing to this branch is very much appreciated :wink:

**Comment by kodebach** [comment]
> `unable to access 'https://github.com/Homebrew/homebrew-core/': transfer closed with outstanding read data remaining` 

Seems like a network error, I restarted the failed builds, let's see what happens...

**Comment by kodebach** [comment]
Seem to work now... (except that Travis also timed out)

**Comment by mpranj** [comment]
Thank you for fixing this so quickly!

**Pull request merged**